packages/core/src/utils.ts

```diff
@@ -92,6 +92,16 @@ export async function copyWalletAddress(text: string): Promise<void> {
   }
 }
 
+export function hexedChain(chains : Chain[]): Chain[] {
+  const hexChain = chains.map(({ id, ...rest }) => { 
+    if(typeof id === 'number'){
+       id = '0x'+Math.abs(id).toString(16)
+      }
+      return { id, ...rest };
+  });
+  return hexChain
+}
+
```

Comment:
```suggestion
export const chainIdToHex = (chain: Chain): Chain => typeof id === 'number' ? { ...chain, id: `0x${chain.id.toString(16)}`} : chain
```
- When possible opt for ternary over multiple returns as I find it more clear/concise
- Template strings are usually preferred over concatenation
- Negative values should just fail so I don't think we need the absolute value
- Also I think that `chainIdToHex` is more reusable and follows similar pattern/naming convention as the object below 👍🏾

Reply:
Hmmn. Thanks. Adapted with a tiny addition.